skimage/filters/thresholding.py

```diff
@@ -766,27 +766,36 @@ def _mean_std(image, w):
            Retrieval XV, (San Jose, USA), Jan. 2008.
            DOI:10.1117/12.767755
     """
-    if w == 1 or w % 2 == 0:
+    if isinstance(w, int):
+        w1 = w
+        w2 = w
+        kern = np.zeros((w2 + 1,) * image.ndim)
+
+    if isinstance(w, tuple) and len(w) == image.ndim:
+        w1 = w[0] * w[1]
+        w2 = max(w)
+        kern = np.zeros(w)
```

Comment:
Here, the kernel should have the shape of the window size (+1 on each dimension, because of the way the integral image works.